### PR TITLE
Remove core .metadata properties and resolveModuleSource

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | 
| Patch: Bug Fix?          | N
| Major: Breaking Change?  | Y
| Minor: New Feature?      | 
| Tests Added/Pass?        | 
| Spec Compliancy?         | 
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  | 

Basically tidying up. Removing:

* Some automated logic to walk the AST for import statements to collect metadata about them, which is useless in most cases since plugins could be injecting things anyway. 
* `resolveModuleSource` which ran on ES6 imports only (unless my remove `babel-helper-module-imports` addition) and kind of useless anyway because it never runs on `require()` calls.
* `usedHelpers` which I can't think of a good usecase for and I don't see a use for.
* `babel.analyse` since it does nothing that `babel.transform` doesn't do.
* `path.mark()` because it, as far as I can tell, something that is super easily to implement in your own plugin, and not used anywhere I can think.

### CODE REVIEW

File: packages/babel-core/src/transformation/store.js [hunk 1]
```diff
@@ -1,15 +0,0 @@
-export default class Store {
```
Comment (hzoo) [inline packages/babel-core/src/transformation/store.js]: yay lol

**Comment by hzoo** [comment]
Doesn't look like the `export function analyze` was removed though?

File: packages/babel-core/src/transformation/file/index.js [hunk 2]
```diff
@@ -95,14 +77,9 @@ export default class File extends Store {
...
-  static helpers: Array<string>;
-
   pluginPasses: Array<Array<[Plugin, Object]>>;
   parserOpts: BabelParserOptions;
   opts: Object;
-  dynamicImportTypes: Object;
-  dynamicImportIds: Object;
-  dynamicImports: Array<Object>;
   declarations: Object;
   usedHelpers: Object;
```
Comment (hzoo) [inline packages/babel-core/src/transformation/file/index.js]: this can be removed

File: packages/babel-core/src/config/removed.js [hunk 3]
```diff
@@ -56,4 +56,15 @@ export default {
   whitelist: {
     message: "Put the specific transforms you want in the `plugins` option",
   },
+
+  resolveModuleSource: {
+    version: 6,
+    message: "Use `babel-plugin-module-resolver@3`'s 'resolvePath' options",
```
Comment (loganfsmyth) [inline packages/babel-core/src/config/removed.js]: cc @tleunen :) Hope that's cool!

**Pull request merged**